lemminx-maven/src/test/java/org/eclipse/lemminx/extensions/maven/participants/hover/ManagedVersionHoverTest.java

```diff
@@ -0,0 +1,103 @@
+import org.eclipse.lsp4j.Hover;
+import org.eclipse.lsp4j.MarkupContent;
+import org.eclipse.lsp4j.Position;
+import org.junit.jupiter.api.AfterEach;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.Timeout;
+import org.junit.jupiter.api.extension.ExtendWith;
+
+@ExtendWith(NoMavenCentralExtension.class)
+public class ManagedVersionHoverTest {
+
+	private XMLLanguageService languageService;
+
+	@BeforeEach
+	public void setUp() throws IOException {
+		languageService = new XMLLanguageService();
+	}
+
+	@AfterEach
+	public void tearDown() throws InterruptedException, ExecutionException {
+		languageService.dispose();
+		languageService = null;
+	}
+
+	@Test
+	@Timeout(90000)
+	public void testManagedVersionHoverInDependencyChiild() throws IOException, URISyntaxException {
+		DOMDocument document = createDOMDocument("/pom-dependencyManagement-child.xml", languageService);
+		Hover hover = languageService.doHover(document, new Position(15, 25), new SharedSettings());
+		String value = ((MarkupContent) hover.getContents().getRight()).getValue();
+		System.out.println("testManagedVersionHoverInDependencyChiild: [" + value + "]");
+		assertNotNull(value);
+		assertTrue(value.contains("The managed version is"));
+		assertTrue(value.contains("2.22.2"));
+		assertTrue(value.contains("The artifact is managed in"));
+		assertTrue(value.contains("org.apache.maven.plugins:maven-surefire-plugin:2.22.2"));
+	}
+
+	@Test
+	@Timeout(90000)
+	public void testManagedVersionHoverInDependencyParent() throws IOException, URISyntaxException {
+		DOMDocument document = createDOMDocument("/pom-dependencyManagement-parent.xml", languageService);
+		Hover hover = languageService.doHover(document, new Position(13, 29), new SharedSettings());
+		String value = ((MarkupContent) hover.getContents().getRight()).getValue();
+		System.out.println("testManagedVersionHoverInDependencyParent: [" + value + "]");
+		assertNotNull(value);
+		assertTrue(value.contains("The managed version is"));
+		assertTrue(value.contains("2.22.2"));
+		assertTrue(value.contains("The artifact is managed in"));
+		assertTrue(value.contains("org.apache.maven.plugins:maven-surefire-plugin:2.22.2"));
+	}
+
+	@Test
+	@Timeout(90000)
+	public void testManagedVersionHoverInPluginChild() throws IOException, URISyntaxException {
+		DOMDocument document = createDOMDocument("/pom-pluginManagement-child.xml", languageService);
+		Hover hover = languageService.doHover(document, new Position(18, 33), new SharedSettings());
+		String value = ((MarkupContent) hover.getContents().getRight()).getValue();
+		System.out.println("testManagedVersionHoverInPluginChild: [" + value + "]");
+		assertNotNull(value);
+		assertTrue(value.contains("The managed version is"));
+		assertTrue(value.contains("2.22.2"));
+		assertTrue(value.contains("The artifact is managed in"));
+		assertTrue(value.contains("org.apache.maven.plugins:maven-surefire-plugin:2.22.2"));
```

Comment:
I think what is expected here is more the reference to the parent pom that adds the version management (here pluginManagement-parent.xml)

Reply:
Fixed